lib/auth/auth.go

```diff
@@ -2554,6 +2554,54 @@ func (a *Server) DeleteNetworkRestrictions(ctx context.Context) error {
 	return a.Services.Restrictions.DeleteNetworkRestrictions(ctx)
 }
 
+// UpsertLock upserts a lock and emits a related audit event.
+func (a *Server) UpsertLock(ctx context.Context, lock types.Lock) error {
+	if err := a.Access.UpsertLock(ctx, lock); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := a.emitter.EmitAuditEvent(a.closeCtx, &apievents.LockCreate{
+		Metadata: apievents.Metadata{
+			Type: events.LockCreatedEvent,
```

Comment:
LockCreatedEvent and LockCreatedCode are slightly misleading, since it could be an update as well. Could we rename those?

Reply:
Yes, I know, but this is a convention we already have in place, see e.g. the `Role`-related events that I largely followed when adding these events.

Reply:
I'd be OK breaking that for clarity's sake, but we might want to ask for a 3rd opinion here. Up to you.